src/screens/TopicPageScreen/elements/BottomSheetFollow.tsx

```diff
@@ -0,0 +1,307 @@
+        paddingHorizontal: dimen.normalizeDimen(20),
+        borderBottomColor: COLORS.gray,
+        borderBottomWidth: 1
+      }}>
+      <View style={{flex: 1, flexDirection: 'column'}}>
+        <View
+          style={{
+            flexDirection: 'row',
+            alignItems: 'center'
+          }}>
+          <View
+            style={{
+              marginRight: 8
+            }}>
+            {type === 'signed' ? (
+              <ProfilePicture
+                size={20}
+                width={0}
+                profilePicPath={profilePicture}
+                anonBackgroundColor=""
+                anonEmojiCode=""
+              />
+            ) : (
+              <IconIncognito color={COLORS.white} />
+            )}
+          </View>
+          <Text
+            style={{
+              fontFamily: fonts.inter[400],
+              fontSize: 14,
+              color: COLORS.black
+            }}>
+            {type === 'signed' ? `as @${username}` : 'Incognito'}
+          </Text>
+        </View>
+        <Text
+          style={{
+            fontFamily: fonts.inter[400],
+            fontSize: 12,
+            color: COLORS.blackgrey,
+            marginTop: 4,
+            lineHeight: 18
+          }}>
+          {type === 'signed'
+            ? 'You’ll be visible as a community member.'
+            : 'You’ll be a hidden member of the community.'}
+        </Text>
+      </View>
+      {isFollowing && (
+        <IconCheckCircleOutline
+          color={type === 'signed' ? COLORS.signed_primary : COLORS.anon_primary}
+        />
+      )}
+    </TouchableOpacity>
+  );
+};
+
+const BottomSheetFollow = forwardRef((props: BottomSheetFollowProps, ref: Ref<RBSheet>) => {
+  const {
+    topicName,
+    memberCount,
+    setMemberCount,
+    isFollow,
+    setIsFollow,
+    followType,
+    setFollowType,
+    getTopicDetail,
+    onClose
+  } = props;
+  const {followTopic} = useChatClientHook();
+  const handleFollowTopic = async ({type}: {type: Follow}) => {
+    if (isFollow) {
+      if (followType === '') {
+        setMemberCount(memberCount - 1);
+      }
+    } else {
+      setMemberCount(memberCount + 1);
+    }
+    onClose();
+    if (type === 'signed') {
+      if (followType === '' || followType === 'incognito') {
+        setFollowType('signed');
+        setIsFollow(true);
+        try {
+          await followTopic(topicName, false);
+          getTopicDetail(topicName);
+        } catch (error) {
+          if (__DEV__) {
+            console.log(error);
+          }
+        }
+      }
+    } else if (type === 'incognito') {
+      if (followType === '' || followType === 'signed') {
+        setFollowType('incognito');
+        setIsFollow(true);
+        try {
+          await followTopic(topicName, true);
+          getTopicDetail(topicName);
+        } catch (error) {
+          if (__DEV__) {
+            console.log(error);
+          }
+        }
+      }
+    }
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `handleFollowTopic` function seems to be handling both the follow and unfollow actions based on the `isFollow` state. This could lead to confusion and potential bugs in the future. It would be better to separate these into two distinct functions, one for following a topic and another for unfollowing. This will make the code more readable and maintainable.

```diff
-  const handleFollowTopic = async ({type}: {type: Follow}) => {
-    if (isFollow) {
-      if (followType === '') {
-        setMemberCount(memberCount - 1);
-      }
-    } else {
-      setMemberCount(memberCount + 1);
-    }
-    onClose();
-    if (type === 'signed') {
-      if (followType === '' || followType === 'incognito') {
-        setFollowType('signed');
-        setIsFollow(true);
-        try {
-          await followTopic(topicName, false);
-          getTopicDetail(topicName);
-        } catch (error) {
-          if (__DEV__) {
-            console.log(error);
-          }
-        }
-      }
-    } else if (type === 'incognito') {
-      if (followType === '' || followType === 'signed') {
-        setFollowType('incognito');
-        setIsFollow(true);
-        try {
-          await followTopic(topicName, true);
-          getTopicDetail(topicName);
-        } catch (error) {
-          if (__DEV__) {
-            console.log(error);
-          }
-        }
-      }
-    }
-  };
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->